package/gluon-setup-mode/files/etc/hotplug.d/button/50-gluon-setup-mode

```diff
@@ -6,6 +6,7 @@ wait=3
 
 wait_setup_mode() {
 	sleep $wait
+	/lib/gluon/setup-mode/led.sh confirm
```

Comment:
There is a race condition here; the shell process could be killed between setting the led and running `gluon-enter-setup-mode`. Maybe we should move the LED trigger into `gluon-enter-setup-mode`?

Reply:
The problem is the target-specific implementation of the `diag.sh` script which throws unrelated errors for our case.

On pure DT platforms we would not need it (only real consumer for us is x86 with apu)

Reply:
I don't understand how this answers my comment

Reply:
Hey, sorry i did not finish it.

So the race condition is for the cause of `gluon-enter-setup-mode` setting `set -eu`.

The LED definition is sourced from `diag.sh` which is included in other contexts where additional variables are set.

Including it within the `gluon-enter-setup-mode` script results in the script aborting early.

Reply:
The race condition I meant was `wait_setup_mode` being killed after `/lib/gluon/setup-mode/led.sh confirm`, but before `gluon-enter-setup-mode`, if the button is released at exactly the wrong moment.

Reply:
Now i get it. I'll think about how to tackle this.

Reply:
The easiest way to fix this would be to call `/lib/gluon/setup-mode/led.sh confirm` from a subprocess together with `gluon-enter-setup-mode` instead of `wait_setup_mode` - either by moving the call into `gluon-enter-setup-mode` itself, or by introducing a new shell using `sh -c` for the two commands.